csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransformLambda.ql

```diff
@@ -0,0 +1,72 @@
+/**
+ * @name Potential usage of a n object implementing ICryptoTransform class in a way that would be unsafe for concurrent threads.
+ * @description An instance of a class that either implements or has a field of type System.Security.Cryptography.ICryptoTransform is being captured by a lambda, 
+ *              and used in what seems to be a thread initialization method.
+ *              Using this an instance of this class in concurrent threads is dangerous as it may not only result in an error, 
+ *              but under some circumstances may also result in incorrect results.
+ * @kind problem
+ * @problem.severity warning
+ * @precision medium
+ * @id cs/thread-unsafe-icryptotransform-captured-in-lambda
+ * @tags concurrency
+ *       security
+ *       external/cwe/cwe-362
+ */
+
+import csharp
+import semmle.code.csharp.dataflow.DataFlow
+
+class ICryptoTransform extends Class {
+  ICryptoTransform() {
+    this.getABaseType*().hasQualifiedName("System.Security.Cryptography", "ICryptoTransform")
+  }
+}
+
+predicate usesICryptoTransformType( Type t ) {
+  exists(  ICryptoTransform ict |
+    ict = t
+    or usesICryptoTransformType( t.getAChild() )
+    or usesICryptoTransformType( t.(Class).getAMember() )
+  )
+}
+
+predicate hasICryptoTransformMember( Class c) {
+  exists( Field f |
+    f = c.getAMember()
+    and ( 
+      exists( ICryptoTransform ict | ict = f.getType() )
```

Comment:
I think this case is redundant since it's covered by `usesICryptoTransformType` below.

Reply:
Fixed